python/ray/_raylet.pyx

```diff
@@ -436,7 +436,14 @@ cdef deserialize_args(
                     == RAW_BUFFER_METADATA):
                 args.append(data)
             else:
-                args.append(pickle.loads(data.to_pybytes()))
+                try:
+                    args.append(pickle.loads(data.to_pybytes()))
+                except Exception:  # XXX handle inline RayObjects properly
```

Comment:
This is dangerous and I have no idea what's happening. Shouldn't this just be a proper by-reference arg?